Source/AwsCommonRuntimeKit/mqtt/MqttRequestResponseClient.swift

```diff
@@ -0,0 +1,299 @@
+///  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+///  SPDX-License-Identifier: Apache-2.0.
+import AwsCMqtt
+import LibNative
+import Foundation
+
+/**
+ * The type of change to the state of a streaming operation subscription
+ */
+public enum SubscriptionStatusEventType {
+    /**
+     * The streaming operation is successfully subscribed to its topic (filter)
+     */
+    case established
+
+    /**
+     * The streaming operation has temporarily lost its subscription to its topic (filter)
+     */
+    case lost
+
+    /**
+     * The streaming operation has entered a terminal state where it has given up trying to subscribe
+     * to its topic (filter).  This is always due to user error (bad topic filter or IoT Core permission
+     * policy).
+     */
+    case halted
+}
+
+extension SubscriptionStatusEventType {
+    /// Returns the native representation of the Swift enum
+    var rawValue: aws_rr_streaming_subscription_event_type {
+        switch self {
+        case .established: return ARRSSET_SUBSCRIPTION_ESTABLISHED
+        case .lost: return ARRSSET_SUBSCRIPTION_LOST
+        case .halted: return ARRSSET_SUBSCRIPTION_HALTED
+        }
+    }
+
+    /// Initializes Swift enum from native representation
+    init(_ cEnum: aws_rr_streaming_subscription_event_type) {
+        switch cEnum {
+        case ARRSSET_SUBSCRIPTION_ESTABLISHED:
+            self = .established
+        case ARRSSET_SUBSCRIPTION_LOST:
+            self = .lost
+        case ARRSSET_SUBSCRIPTION_HALTED:
+            self = .halted
+        default:
+            fatalError("Unknown Susbscription Event Type")
+        }
+    }
+}
+
+/// An event that describes a change in subscription status for a streaming operation.
+public struct SubscriptionStatusEvent {
+    /// The type of the event
+    public let event: SubscriptionStatusEventType
+    
+    /// An optional error code associated with the event. Only set for SubscriptionLost and SubscriptionHalted.
+    public let error: CRTError?
+}
+
+// TODO: Igor has updated the events for IoT Command. Need update later
+/// An event that describes an incoming publish message received on a streaming operation.
+public struct IncomingPublishEvent {
+
+    /// The payload of the publish message in a byte buffer format
+    let payload: Data
+
+    /// The topic associated with this PUBLISH packet.
+    let topic: String
+    
+    // TODO: More options for IoT Command changes
+}
+
+/// Function signature of a SubscriptionStatusEvent event handler
+public typealias SubscriptionStatusEventHandler = (SubscriptionStatusEvent) async -> Void
```

Comment:
In `StreamingOperationOptions` and its eventual `init()`, these async callbacks will probably need to be `@escaping` so we might want to go ahead and make these explicitly `@Sendable` here

Reply:
Why are they async to begin with?  They should not be